dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java

```diff
@@ -292,22 +286,22 @@ public Response filter(@Context final HttpServletRequest request, @PathParam("pa
 	 * 
 	 * @param request
 	 *            - The {@link HttpServletRequest} object.
-	 * @param params
 	 *            - The parameters that can be specified in the REST call.
 	 * @return A {@link Response} containing the status of the operation. This
 	 *         will probably require a page refresh.
 	 * @throws Exception An error occurred when authenticating the request.
 	 */
-	@PUT
-	@Path("/loginas/{params:.*}")
+	@POST
+	@Path("/loginas")
 	@JSONP
 	@NoCache
 	@Produces({ MediaType.APPLICATION_JSON, "application/javascript" })
-	public final Response loginAs(@Context final HttpServletRequest request, @PathParam("params") final String params) throws Exception {
-		InitDataObject initData = webResource.init(params, true, request, true, null);
-		final Map<String, String> urlParams = initData.getParamsMap();
-		final String loginAsUserId = getMapValue(urlParams, "userid", null);
-		final String loginAsUserPwd = getMapValue(urlParams, "pwd", null);
+	public final Response loginAs(@Context final HttpServletRequest request, final LoginAsForm loginAsForm) throws Exception {
+		InitDataObject initData = webResource.initWithMap(CollectionsUtils.map("userid", loginAsForm.getUserId(), "pwd", loginAsForm.getPassword()),
```

Comment:
The initWithMap should be internal (private) provide a init (user, password)
under the method can call the initWithMap, the implementation will be easier to understand, more than just a map